packages/binary/messagelayer/tipselector/events.go

```diff
@@ -14,6 +14,13 @@ type Events struct {
 	TipRemoved *events.Event
 }
 
+func newEvents() *Events {
+	return &Events{
+		TipAdded:   events.NewEvent(messageIDEvent),
+		TipRemoved: events.NewEvent(messageIDEvent),
+	}
+}
+
 func messageIDEvent(handler interface{}, params ...interface{}) {
 	handler.(func(message.ID))(params[0].(message.ID))
 }
```

Comment:
Should that be embedded in a struct? Or does it overcomplicate things?

Reply:
yeah, for those events that are rather simple, like `messageIDEvent` should and only contains `message.ID`, we could just keep the way it was.